internal/pkg/db/db.go

```diff
@@ -93,14 +101,81 @@ func (backend *Backend) AddDocument(document *sbom.Document) error {
 func (backend *Backend) GetDocumentByID(id string) (*sbom.Document, error) {
 	document, err := backend.Retrieve(id, nil)
 	if err != nil {
-		backend.Logger.Warn("Document could not be retrieved", "id", id, "err", err)
+		backend.Logger.Debug("Document could not be retrieved", "id", id, "err", err)
 
 		return nil, fmt.Errorf("failed to retrieve document: %w", err)
 	}
 
 	return document, nil
 }
 
+func (backend *Backend) GetDocumentByIDOrAlias(id string) (*sbom.Document, error) {
+	document, idErr := backend.GetDocumentByID(id)
+
+	if document == nil {
+		documents, aliasErr := backend.GetDocumentsByAnnotation(BomctlAnnotationAlias, id)
```

Comment:
Rename to just `err`

Reply:
If I do that with both errors, then it will be shadowing the ID error in the outer scope. I think we should capture both in the log.

Reply:
The first error would have been returned due to the critical database failure

Reply:
I renamed the error,  but `backend.GetDocumentByID` is not throwing an error just because of critical database failure. It also errors out if the ID is not found. I'll open an issue for that in `protobom-storage`.